src/engraving/dom/undo.h

```diff
@@ -1511,6 +1511,28 @@ class ChangeDrumset : public UndoCommand
     UNDO_NAME("ChangeDrumset")
 };
 
+class FretDataChange : public UndoCommand
+{
+    OBJECT_ALLOCATOR(engraving, FretDataChange)
+
+    FretDiagram* m_diagram = nullptr;
+    FretUndoData m_undoData;
+    String m_harmonyName;
+
+    void redo(EditData*) override;
+    void undo(EditData*) override;
+
+public:
+    FretDataChange(FretDiagram* d, const String& harmonyName)
```

Comment:
I'd suggest renaming it to FretHarmonyChange or ChangeFretHarmony

Reply:
but it doesn't change harmony, it changes only diagram data